pennylane/measurements/sample.py

```diff
@@ -106,4 +109,49 @@ def circuit(x):
             )
         wires = Wires(wires)
 
-    return MeasurementProcess(Sample, obs=op, wires=wires)
+    return _Sample(Sample, obs=op, wires=wires)
+
+
+# TODO: Make public when removing the ObservableReturnTypes enum
+class _Sample(SampleMeasurement):
+    """Measurement process that returns the samples of a given observable."""
+
+    def process_samples(
+        self, samples: Sequence[complex], shot_range: Tuple[int] = None, bin_size: int = None
+    ):
+        name = self.obs.name if self.obs is not None else None
+        # Select the samples from samples that correspond to ``shot_range`` if provided
+        if shot_range is not None:
+            # Indexing corresponds to: (potential broadcasting, shots, wires). Note that the last
+            # colon (:) is required because shots is the second-to-last axis and the
+            # Ellipsis (...) otherwise would take up broadcasting and shots axes.
+            samples = samples[..., slice(*shot_range), :]
+
+        if len(self.wires) != 0:
+            # if wires are provided, then we only return samples from those wires
+            samples = samples[..., self.wires]
```

Comment:
I think we'll need to map the wires here too.

Reply:
You are completely right! Added the `wire_order` argument to the `process_samples` function.

Also changed the name from `wires` to `wire_order` in the `process_state` function for more clarity.